projects/RabbitMQ.Client/client/impl/ConsumerDispatching/AsyncConsumerDispatcher.cs

```diff
@@ -26,22 +27,56 @@ protected override async Task ProcessChannelAsync(CancellationToken token)
                         {
                             try
                             {
-                                Task task = work.WorkType switch
+                                Task task = Task.CompletedTask;
+                                switch (work.WorkType)
                                 {
-                                    WorkType.Deliver => work.AsyncConsumer.HandleBasicDeliver(
-                                        work.ConsumerTag, work.DeliveryTag, work.Redelivered,
-                                        work.Exchange, work.RoutingKey, work.BasicProperties, work.Body.Memory),
+                                    case WorkType.Deliver:
+                                        {
+                                            IAsyncBasicConsumer? deliverConsumer = work.AsyncConsumer ??
+                                                throw new InvalidOperationException("[CRITICAL] should never see this");
+                                            ConsumerTag? deliverConsumerTag = work.ConsumerTag;
+                                            var method = new BasicDeliver(work.Method.Memory);
```

Comment:
I do not see where the rented method memory is released again? Have I missed it somehwere?

Reply:
(Applies to everywhere, not specifically to this call here)

Reply:
https://github.com/zgabi/rabbitmq-dotnet-client/blob/stringallocations/projects/RabbitMQ.Client/client/impl/ConsumerDispatching/AsyncConsumerDispatcher.cs#L26-L86

The `WorkStruct` is disposed at the end of this `using` block:

https://github.com/zgabi/rabbitmq-dotnet-client/blob/stringallocations/projects/RabbitMQ.Client/client/impl/ConsumerDispatching/AsyncConsumerDispatcher.cs#L26

Reply:
Don't get me wrong, I am not thrilled with how convoluted the code is in this library, but it is what it is at this time.

Reply:
Ah thanks, missed that using, as it was already like that before the change, so it didn't show up 👍

Reply:
No problem! It is not intuitive **at all**